google/cloud/sql/connector/utils.py

```diff
@@ -14,13 +14,25 @@
 limitations under the License.
 """
 
+import asyncio
+from functools import partial, wraps
 
 import pymysql.cursors
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import serialization
 from cryptography.hazmat.primitives.asymmetric import rsa
 
 
+def run_function_as_async(func):
+    @wraps(func)
+    async def wrapped_sync_function(*args, **kwargs):
+        partial_func = partial(func, *args, **kwargs)
+        loop = kwargs.pop("loop") or asyncio.get_event_loop()
```

Comment:
So after looking back at this code with more context, I think this was better without the explicit loop. It should only be used from an asyncio context, and giving the ability to pass a loop just introduces a chance for mistakes to happen.